backend/internal/handlers/friendship/friendship.go

```diff
@@ -1,28 +1,43 @@
 package friendship
 
 import (
-	"fmt"
 	"github.com/GenerateNU/nightlife/internal/models"
 	"github.com/gofiber/fiber/v2"
-	"log"
+	"github.com/google/uuid"
 )
 
 // POST endpoint to create friendship between two users in DB
 func (s *Service) CreateFriendship(c *fiber.Ctx) error {
-	fmt.Println("Creating a friendship")
+    var req models.Friendship
 
-	var req models.Friendship
+    // Parse the request body and check for errors
+    if err := c.BodyParser(&req); err != nil {
+        return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{
+            "error": "Cannot parse JSON",
+        })
+    }
 
-	if err := c.BodyParser(&req); err != nil {
-		log.Printf("Error parsing JSON: %v, Request: %+v", err, req)
-		return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{
-			"error": "Cannot parse JSON",
-		})
-	}
+    // Ensure both user ids are present
+    if req.UserID1 == uuid.Nil || req.UserID2 == uuid.Nil {
+        return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{
+            "error": "Both user_id1 and user_id2 are required",
+        })
+    }
 
-	if err := s.store.CreateFriendship(c.Context(), req); err != nil {
-		return err
-	}
+    // Set default friendship status if not provided (just an edge case)
+    if req.FriendshipStatus == "" {
+        req.FriendshipStatus = models.Pending
+    }
 
-	return c.Status(fiber.StatusCreated).JSON(req)
+    // Call CreateFriendship method to interact with db
+    if err := s.store.CreateFriendship(c.Context(), req); err != nil {
+        return c.Status(fiber.StatusInternalServerError).JSON(fiber.Map{
```

Comment:
with the central error handler, can just bubble up an `errors.New("failed to create friendship")`